lib/subset.cc

```diff
@@ -1350,20 +1355,22 @@ void SubsetPartition::merge_from_disk(string other_filename)
             diskp = (PartitionID *) (buf + i);
             i += sizeof(PartitionID);
 
-            if (!(*diskp != 0)) {		// sanity check.
-                throw khmer_exception();
-            }
+            assert((*diskp != 0)); // sanity check!
```

Comment:
On Thu, Jun 04, 2015 at 06:38:49PM -0700, Camille Scott wrote:
> > @@ -1350,20 +1355,22 @@ void SubsetPartition::merge_from_disk(string other_filename)
> >              diskp = (PartitionID *) (buf + i);
> >              i += sizeof(PartitionID);
> >  
> > -            if (!(*diskp != 0)) {		// sanity check.
> > -                throw khmer_exception();
> > -            }
> > +            assert((*diskp != 0)); // sanity check!
> 
> Thought we were trying to avoid `assert`s in the C++ code?

I was unable to think of a way to trigger this in a test, and this is a
reasonable check on things when debugging or modifying other code.  I think
we're trying to avoid asserts as things that are actually raised for users, but
I don't see any reason not to leave them in as sanity checks.